packages/taquito/src/operations/types.ts

```diff
@@ -563,6 +566,53 @@ export interface FailingNoopParams {
   basedOnBlock: BlockIdentifier;
 }
 
+export interface InternalStakingParams {
+  source: string;
+  amount: number;
+  fee?: number;
+  gasLimit?: number;
+  storageLimit?: number;
+  mutez?: boolean;
+  parameter: TransactionOperationParameter;
+}
+
+type PartialBy<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>;
+export type StakingParams = PartialBy<InternalStakingParams, 'source'>;
+export type StakeParams = Omit<StakingParams, 'parameter'>;
```

Comment:
preferably having each params defined by themselves explicitly for readability and better user experience

Reply:
I think that causes worse maintainability. Also it is WET (Write Everything Twice) as opposed to DRY (Don't Repeat Yourself)